src/libraries/Native/Unix/System.Native/pal_networkchange.c

```diff
@@ -117,3 +131,44 @@ void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)
         }
     }
 }
+#elif HAVE_RT_MSGHDR
+void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)
+{
+    char buffer[4096];
```

Comment:
I'm wondering if we should have some #define here but we already use number in Linux case. 
Do you know what would happen if we ever read incomplete message @am11 ?  
4k seems enough by looking at `rtsock_msg_buffer()`.

Reply:
>  Do you know what would happen if we ever read incomplete message

I have not experienced this case so i cannot say for sure.
Based on the fact that all four possible types of headers `rt_msghdr`, `if_msghdr`, `ifa_msghdr` and `if_announcemsghdr` have fixed offset for message type, it seems like in case of incomplete read, after the version validation condition has passed (which narrows down the chances of corruption), we will likely misread garbage as `rtm_type` (1 byte) if that was the breakage point. The valid range of values for this type byte is: 0x1:0x18 and we only care about a subset of it. Thus, if chances of getting invalid notification is there, it is very low. If that was the last message in the pipe, this loop will exit, and on managed side we create a new socket.